airdcpp/airdcpp/PrivateChat.cpp

```diff
@@ -437,17 +437,19 @@ void PrivateChat::on(AdcCommand::PMI, UserConnection*, const AdcCommand& cmd) no
 		fire(PrivateChatListener::PMStatus(), this, type);
 }
 
-void PrivateChat::logMessage(const string& aMessage) const noexcept {
+void PrivateChat::logMessage(const ChatMessagePtr& aMessage) const noexcept {
 	if (SETTING(LOG_PRIVATE_CHAT)) {
 		ParamMap params;
-		params["message"] = aMessage;
+		params["message"] = aMessage->format();
+		params["userI4"] = aMessage->getFrom()->getIdentity().getIp4();
+		params["2code"] = aMessage->getFrom()->getIdentity().getCountry();
```

Comment:
Currently all identity-related are prefixed with `my`/`hub`/`user`, where `user` refers to the session user. It might make sense to follow the same logic also in here. Maybe the prefix for these params could `from` to avoid confusion...

I'd also prefer fetching the country only if the param is actually being used (similar to what  `fillLogParams` does), even though it may not make much difference performance-wise with the current country lookup implementation.

Is there a specific reason why you don't want to add a param for I6, which has an identical getter function?

Reply:
Hmm. For me "user" is other user (the one who sent message), "my" is myself and "hub" is hub. If we follow your logic then "userNI" should became "fromNI" in private chats aswell.

About "2code", yes, I was thinking that when I was implementing the variables. In my ApexDC mod I use "userCC", "userCN", "userASN" etc.

I'm working on IPv6, that is in to do list. :)

Reply:
Right, about IPv6. This would break the functionality for IPv6 hubs, since we can't log each hub separately. So basically a global variable is needed there, either "userIP" or "ip" for compatibility.